android/cli/commands/_buildModule.js

```diff
@@ -293,6 +283,76 @@ AndroidModuleBuilder.prototype.initialize = function initialize(next) {
 	this.sharedHooksDir = path.resolve(this.projectDir, '..', 'hooks');
 
 	this.timoduleXmlFile = path.join(this.projectDir, 'timodule.xml');
+	this.timodule = fs.existsSync(this.timoduleXmlFile) ? new tiappxml(this.timoduleXmlFile) : undefined;
+	this.modulesDir = path.join(this.projectDir, 'modules', 'android');
+	this.globalModulesDir = path.join(this.globalModulesPath, 'android');
+
+	// process module dependencies
+	this.modules = this.timodule && !Array.isArray(this.timodule.modules) ? [] : this.timodule.modules.filter(function(m) {
+		if (!m.platform || /^android$/.test(m.platform)) {
+			var localPath = path.join(this.modulesDir, m.id),
+				globalPath = path.join(this.globalModulesDir, m.id),
+				getModulePath = function(modulePath) {
+					var items = fs.readdirSync(modulePath);
+					if (m.version) {
+						for (var item of items) {
+							if (item === m.version) {
+								m.path = path.join(modulePath, m.version);
+								return true;
+							}
+						}
+					} else if (items.length) {
+						var latest = items[items.length - 1];
+						if (!latest.startsWith('.')) {
+							m.version = latest;
+							m.path = path.join(modulePath, m.version);
+							return true;
+						}
+					}
+					return false;
+				}.bind(this);
+
+			if ((fs.existsSync(localPath) && getModulePath(localPath)) ||
+				(fs.existsSync(globalPath) && getModulePath(globalPath))) {
+				return true;
+			}
+		}
+		return false;
+	}.bind(this));
+
+	// obtain module dependency android archives for aar-transform to find
+	this.moduleAndroidLibraries = [];
+	for (var module of this.modules) {
+		var libPath = path.join(module.path, 'lib');
+		fs.existsSync(libPath) && fs.readdirSync(libPath).forEach(function (name) {
+			var file = path.join(libPath, name);
+			if (/\.aar$/.test(name) && fs.existsSync(file)) {
+				this.moduleAndroidLibraries.push({
+					aarPathAndFilename: String(file),
+					originType: 'Module'
+				});
+			}
+		}, this);
+	}
+
+	// module java archive paths
+	this.jarPaths = [ path.join(this.platformPath, 'lib'), path.join(this.platformPath, 'modules'), this.platformPath ];
+
+	// module dependencies java archive paths
+	for (var module of this.modules) {
```

Comment:
This should not be necessary as the aar hook already loops over all .jar files here: https://github.com/garymathews/titanium_mobile/blob/ac64609be01f8caeeb5baf68c760fe8350bdcd45/android/cli/hooks/aar-transform.js#L305